model/domain/signedpiece.go

```diff
@@ -120,3 +142,15 @@ func (sp *SignedPiece) Verify() (pieceCid string, err error) {
 
 	return pieceCid, nil
 }
+
+func decodeHex(src []byte) ([]byte, error) {
+	src = bytes.TrimPrefix(src, []byte("0x"))
+	decoded := make([]byte, hex.DecodedLen(len(src)))
+	_, err := hex.Decode(decoded, src)
+	return decoded, err
+}
+
+// Format the time the same way as JavaScript Date.toISOString()
+func formatTimestamp(unixMilli uint64) string {
+	return time.UnixMilli(int64(unixMilli)).UTC().Format("2006-01-02T15:04:05.000Z")
```

Comment:
Did you try default golang formats `time.RFC3339` or `time.RFC3339Nano`? Probably one of these formats can parse.

Reply:
That's right, it's just harder to implement in pure Javascript. Maybe I'll change it anyway.

Reply:
No I’ll keep it, it maintains exactly the milliseconds and without timezone problems.